src/cc/arduino/plugins/wifi101/UpdaterImpl.java

```diff
@@ -27,65 +27,105 @@
  */
 package cc.arduino.plugins.wifi101;
 
+import java.io.File;
+import java.net.URISyntaxException;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 
+import static java.util.Arrays.asList;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+
+import javax.swing.DefaultListModel;
 import javax.swing.JOptionPane;
 
 import cc.arduino.packages.BoardPort;
-import cc.arduino.plugins.wifi101.firmwares.WINC1500Firmware;
 import cc.arduino.plugins.wifi101.flashers.Flasher;
-import cc.arduino.plugins.wifi101.flashers.java.JavaFlasher;
+import cc.arduino.plugins.wifi101.flashers.java.WINCFlasher;
+import cc.arduino.plugins.wifi101.flashers.java.NinaFlasher;
 import processing.app.Base;
 
 @SuppressWarnings("serial")
 public class UpdaterImpl extends UpdaterJFrame {
+	private SerialPortListModel listModel;
+	private List<String> websites = new ArrayList<>();
 
-	private Flasher flasher;
+	public ArrayList<String> compatibleBoard;
+	private HashMap <String,String> boardList;
+
+	public static ArrayList<Flasher> fwAvailable = new ArrayList<Flasher>();
 
 	public UpdaterImpl() throws Exception {
 		super();
+		boardList = new HashMap<String,String>();
 		Base.registerWindowCloseKeys(getRootPane(), e -> {
 			setVisible(false);
 		});
 		Base.setIcon(this);
 
-		for (WINC1500Firmware firmware : WINC1500Firmware.available)
-			getFirmwareSelector().addItem(firmware);
+		fwAvailable.add(new NinaFlasher("NINA firmware", "1.0.0", "firmwares/NINA/1.0.0/nina-fw.bin", new ArrayList<String>(asList("Arduino MKR WiFi 1010", "Arduino MKR Vidor 4000", "Arduino UNO WiFi REV2"))));
+		fwAvailable.add(new WINCFlasher("WINC1501 Model B", "19.5.4", "firmwares/WINC1500/19.5.4/m2m_aio_3a0.bin", new ArrayList<String>(asList("Arduino/Genuino MKR1000"))));
+		fwAvailable.add(new WINCFlasher("WINC1501 Model B", "19.5.2", "firmwares/WINC1500/19.5.2/m2m_aio_3a0.bin", new ArrayList<String>(asList("Arduino/Genuino MKR1000"))));
+		fwAvailable.add(new WINCFlasher("WINC1501 Model B", "19.4.4", "firmwares/WINC1500/19.4.4/m2m_aio_3a0.bin", new ArrayList<String>(asList("Arduino/Genuino MKR1000"))));
+		fwAvailable.add(new WINCFlasher("WINC1501 Model A", "19.4.4", "firmwares/WINC1500/19.4.4/m2m_aio_2b0.bin", new ArrayList<String>(asList("Arduino/Genuino MKR1000", "Wifi 101"))));
 
-		refreshSerialPortList();
+		boardList.put("2341:8054", "NINA");
+		boardList.put("2341:8056", "NINA");
+		boardList.put("03eb:2145", "NINA");
+		boardList.put("2341:804e", "WINC1500");
```

Comment:
Can we fetch the USB id's via some classes from the IDE?

Reply:
Removed the code work without this hashmap